src/internal/transactionenv/context/context.go

```diff
@@ -0,0 +1,69 @@
+package txncontext
+
+import (
+	"context"
+
+	col "github.com/pachyderm/pachyderm/v2/src/internal/collection"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+// TransactionContext is a helper type to encapsulate the state for a given
+// set of operations being performed in the Pachyderm API.  When a new
+// transaction is started, a context will be created for it containing these
+// objects, which will be threaded through to every API call:
+//   ctx: the client context which initiated the operations being performed
+//   pachClient: the APIClient associated with the client context ctx
+//   stm: the object that controls transactionality with etcd.  This is to ensure
+//     that all reads and writes are consistent until changes are committed.
+//   txnEnv: a struct containing references to each API server, it can be used
+//     to make calls to other API servers (e.g. checking auth permissions)
+//   pfsDefer: an interface for ensuring certain PFS cleanup tasks are performed
+//     properly (and deduped) at the end of the transaction.
+type TransactionContext struct {
```

Comment:
Split `TransactionContext` out of `transactionenv` to avoid a circular dependency between the API definitions and the transactionenv package.